doc/phar-signing.md

```diff
@@ -0,0 +1,364 @@
+```
+
+That will leave the public and private key in a single file. Anyone that has that file can sign on your behalf! So keep
+that file secure at all times and make sure it never accidentally shows up in your git repository!
+
+
+### Secure your encryption key
+
+If your goal is to save this encryption key somewhere, for example your repository, you should first encrypt it:
+
+```
+gpg --symmetric keys.asc
+```
+
+This will ask for a second passphrase. It is recommended to **pick a different passphrase** than for the key itself and
+ideally one generated and managed by a password manager.
+
+This leaves you with a file `keys.asc.gpg`. You can add this one to the repository and at this point you are probably
+better off **deleting the `keys.asc` file**. In order to do the actual signing, you will have to decrypt it again, but
+it is better to not keep that decrypted key around. 
+
+
+### Sign your PHAR
+
+You first need to encrypt `keys.asc.gpg` into `keys.asc`:
+
+```
+# If you are locally:
+$ gpg keys.asc.gpg
+# In another environment: CI or other. You should use an environment variable
+# or a temporary file to avoid printing the password in clear text.
+echo $DECRYPT_KEY_PASSPHRASE | gpg --passphrase-fd 0 keys.asc.gpg
+# or:
+cat $(.decrypt-key-passphrase) | gpg --passphrase-fd 0 keys.asc.gpg
+```
+
+Import the decrypted key if it is not already present on the machine:
+
+```
+$ gpg --batch --yes --import keys.asc
+```
+
+Sign your file:
+
+```
+$ gpg \
+   --batch \
+   --passphrase="$GPG_KEY_96C8013A3CC293C465EE3FBB03B2F4DF7A20DF08_PASSPHRASE" \
+   --local-user 96C8013A3CC293C465EE3FBB03B2F4DF7A20DF08 \
+   --armor \
+   --detach-sign \
+   bin/command.phar
+# Do not forget to remove keys.asc afterwards!
+```
+
+You will now have a file `bin/command.phar.asc`.
+
+When publishing your archive, you should publish both `bin/command.phar` and `bin/command.phar.asc`.
+
+
+### Verifying the PHAR signature
+
+First you should check the issuer's identity, usually it is provided from where you download it as part of the
+documentation:
+
+```
+gpg --keyserver hkps://keys.openpgp.org --recv-keys 96C8013A3CC293C465EE3FBB03B2F4DF7A20DF08
+```
+
+However not everyone expose what is their GPG key ID. So sometimes to avoid bad surprises, you
+can look up for similar issuers to the key ID given by the `.asc`:
+
+```
+# Verify the signature
+$ gpg --verify bin/command.phar.asc bin/command.phar
+
+# Example of output:
+gpg: Signature made Sat 21 Oct 16:58:05 2023 CEST
+gpg:                using EDDSA key 96C8013A3CC293C465EE3FBB03B2F4DF7A20DF08
+gpg: Good signature from "Théo Fidry <theo.fidry+phar-signing-example@example.com>" [ultimate]
+```
+
+If the key ID was not provided before, you can try to look it up to check it was properly registered to a keyserver:
+
+```
+$ gpg --keyserver https://keys.openpgp.org --search-keys "theo.fidry+phar-signing-example@example.com"
+```
+
```

Comment:
It might be a nice addition to mention [phive](https://phar.io/) here, as that does all that automagically.